docs/design/diagrams/generate.py

```diff
@@ -0,0 +1,112 @@
+from diagrams import Cluster, Diagram, Edge
+from diagrams.onprem.database import PostgreSQL
+from diagrams.onprem.inmemory import Redis
+from diagrams.k8s.controlplane import API
+from diagrams.custom import Custom
+from diagrams.onprem.client import Client
+
+graph_attr = {
+    "concentrate": "false",
+    "splines": "ortho",
+    "pad": "0.5",
+    "nodesep": "0.60",
+    "ranksep": "1.5",
+}
+
+edge_attr = {
+    "minlen": "1",
+}
+
+cluster_attr_common = {
+    "margin": "20",
+}
+
+cluster_attr_armada = {
+    "labelloc": "b",
+    "margin": "1",
+}
+cluster_attr_armada.update(cluster_attr_common)
+
+
+armada_logo = "./images/armada.png"
+pulsar_logo = "./images/pulsar.png"
+browser_logo = "./images/browser.png"
+
+with Diagram(
+    name="Armada Systems Diagram",
+    show=False,
+    direction="BT",
+    graph_attr=graph_attr,
+    edge_attr=edge_attr,
+    filename="armada-system",
+):
+
+    with Cluster("External Services", graph_attr=cluster_attr_common):
+        pulsar = Custom("Pulsar", pulsar_logo)
+        postgres = PostgreSQL("Postgres")
+        redis = Redis("Redis")
+
+    armada_clients = Client("Armada Clients")
+
+    with Cluster("Armada Server Components", graph_attr=cluster_attr_armada):
+
+        with Cluster(
+            "Armada Server Cluster", graph_attr=cluster_attr_armada
+        ) as services:
+
+            lookout_api = Custom("Lookout API", armada_logo)
+
+            lookout_ingester = Custom("Lookout Ingester", armada_logo)
+            event_ingester = Custom("Event Ingester", armada_logo)
+
+            armada_server = Custom("Armada Server", armada_logo)
+
+        with Cluster(
+            "Armada Executor Cluster", graph_attr=cluster_attr_armada
+        ) as workers:
+
+            armada_executor = Custom("Armada Executor", armada_logo)
+            binoculars = Custom("Binoculars", armada_logo)
+
+    k8s_api = API("K8s API")
```

Comment:
Shouldn't this be an array?

Reply:
Sorry, can you explain this a bit more?

Reply:
Don't we talk to multiple k8s api instances? IE, the whole point of the thing :)

Reply:
Yeah, I was thinking about this: I can definitely make that more clearly, put for the purposes of these components connect, I felt that is kind of abstracted away? 

In hindsight, I definitely lean more towards the former, I'll get that done :))